app/controllers/controllers.js

```diff
@@ -201,58 +201,74 @@ exports.showAllTeams = (req, res) => {
 
 // Handles teammate evaluation
 exports.evaluateTeammate = async (req, res) => {
-  if (req.session.user) {
-    var teammateID = req.params.id;
-    var reviewerID = req.session.user.id;
+  // Check if user is logged in (session should exist)
+  if (!req.session.user || !req.session.user.id) {
+    return res.redirect("/login"); // Redirect to login page if user is not logged in
+  }
+
+  const teammateID = req.params.id;
+  const reviewerID = req.session.user.id;
 
-    var sql =
-      `SELECT * FROM evaluations WHERE teammateID = ${teammateID} ` +
-      `AND reviewerID = ${reviewerID}`;
+  console.log("Teammate ID:", teammateID);
+  console.log("Reviewer ID:", reviewerID);
 
+  if (!teammateID || !reviewerID) {
+    return res.status(400).send("Invalid IDs provided.");
+  }
+
+  try {
+    // Query to check if the current reviewer has already evaluated the teammate
+    let sql = `
+      SELECT 
+        TypeOfEval, score, comments
+      FROM evaluations 
+      WHERE teammateID = ${teammateID} AND reviewerID = ${reviewerID}
+    `;
     const result = await db.query(sql);
 
-    // Checks if an evaluation has been submitted already
-    if (result[0].length > 0) {
-      res.render("ViewEvaluation.ejs", { review: result[0][0] });
-    } else {
```

Comment:
you should ensure that both result and result[0] are valid before accessing .length